sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/impl/action/ActionsImpl.scala

```diff
@@ -83,10 +85,26 @@ final class ActionService(
 private[javasdk] object ActionsImpl {
   private[action] val log = LoggerFactory.getLogger(classOf[ActionsImpl])
 
-  private def handleUnexpectedException(service: ActionService, command: ActionCommand, ex: Throwable): ActionResponse =
+  private def handleUnexpectedException(
+      service: ActionService,
+      command: ActionCommand,
+      ex: Throwable): ActionResponse = {
+    ex match {
+      case badReqEx: BadRequestException => handleBadRequest(service, command, badReqEx)
+      case _ =>
+        ErrorHandling.withCorrelationId { correlationId =>
+          service.log.error(s"Failure during handling of command ${command.serviceName}.${command.name}", ex)
+          protocolFailure(correlationId)
+        }
+    }
+  }
+
+  private def handleBadRequest(service: ActionService, command: ActionCommand, ex: Throwable): ActionResponse =
     ErrorHandling.withCorrelationId { correlationId =>
-      service.log.error(s"Failure during handling of command ${command.serviceName}.${command.name}", ex)
-      protocolFailure(correlationId)
+      service.log.error(s"Bad request when handling of command ${command.serviceName}.${command.name}", ex)
```

Comment:
Should be an error log, it's a malformed request from the user and should say what is wrong in the response to the user instead.

Reply:
Yes, validation errors in HTTP (and gRPC) APIs should generally not lead to any logging on the server side, the error is at the client side so needs to go in response so the client can fix their request.

Reply:
Ok, so it `Shouldn't` :)

Reply:
Ah, runaway `n`t`, sorry didn't notice.